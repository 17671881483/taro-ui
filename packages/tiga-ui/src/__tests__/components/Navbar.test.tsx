import React from 'react'
import renderer from 'react-test-renderer'
import { cleanup, fireEvent, render } from '@testing-library/react'

describe('AtNavBar 测试在 h5 环境，rn类似 -> ', () => {
  const OLD_ENV = { ...process.env }
  process.env.TARO_ENV = 'weapp'

  afterAll(() => {
    process.env = { ...OLD_ENV } // Restore old environment
    jest.resetModules()
  })

  afterEach(cleanup)

  const AtNavBar = jest.requireActual('../../components/navbar').default

  it('展示：默认导航栏', () => {
    const component = renderer.create(<AtNavBar title='自定义标题' />)

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('展示：左边自定义（最多四个字）', () => {
    const component = renderer.create(<AtNavBar title='标题' leftIcon='取消' />)

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('展示：右边自定义（最多四个字）', () => {
    const component = renderer.create(
      <AtNavBar title='标题' extra='右边内容' />
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('交互：点击左侧区域', () => {
    const onBackFn = jest.fn()

    const { getByText } = render(
      <AtNavBar title='标题' leftIcon='返回' onBack={onBackFn} />
    )

    fireEvent.click(getByText('返回'))
    expect(onBackFn).toBeCalledTimes(1)
  })
})
