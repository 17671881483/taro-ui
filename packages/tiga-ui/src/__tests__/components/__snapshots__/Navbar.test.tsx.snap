// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtNavBar 测试在 h5 环境，rn类似 ->  展示：右边自定义（最多四个字） 1`] = `
<view
  className="at-nav-bar"
>
  <view
    className="at-nav-bar__back"
    onClick={[Function]}
  >
    <text
      className="at-icon at-icon-close"
      style={
        Object {
          "color": "",
          "fontSize": "18px",
        }
      }
    />
  </view>
  <view
    className="at-nav-bar__title"
  >
    标题
  </view>
  <view
    className="at-nav-bar__extra"
  >
    右边内容
  </view>
</view>
`;

exports[`AtNavBar 测试在 h5 环境，rn类似 ->  展示：左边自定义（最多四个字） 1`] = `
<view
  className="at-nav-bar"
>
  <view
    className="at-nav-bar__back"
    onClick={[Function]}
  >
    取消
  </view>
  <view
    className="at-nav-bar__title"
  >
    标题
  </view>
  <view
    className="at-nav-bar__extra"
  />
</view>
`;

exports[`AtNavBar 测试在 h5 环境，rn类似 ->  展示：默认导航栏 1`] = `
<view
  className="at-nav-bar"
>
  <view
    className="at-nav-bar__back"
    onClick={[Function]}
  >
    <text
      className="at-icon at-icon-close"
      style={
        Object {
          "color": "",
          "fontSize": "18px",
        }
      }
    />
  </view>
  <view
    className="at-nav-bar__title"
  >
    自定义标题
  </view>
  <view
    className="at-nav-bar__extra"
  />
</view>
`;
